src/processors/models/database/sequelize/sequelizeConnector.js

```diff
@@ -54,6 +55,12 @@ async function deleteProcessor(processorId) {
     });
 }
 
+async function updateProcessor(processorId, updatedProcessor) {
+    const oldProcessor = await getProcessor(processorId);
```

Comment:
instead of fetching the old processor and then updating it, use the processor sequelize model `const processor = client.model('processor')` and run an update query with `where` clause:

`await processor.update(params, { where: { id: $ID })`

Reply:
done